### PR TITLE
Data handling updates

### PR DESCRIPTION
**Pull request by koenderks**

### CODE REVIEW

**Review by vandenman — CHANGES_REQUESTED** [review]
looks good, some minor things below.

File: R/bainCommon.R [hunk 1]
```diff
@@ -52,23 +52,17 @@
...
-        }
+    dataset[factors] <- lapply(dataset[factors], as.factor)
+    missing <- names(which(apply(dataset, 2, function(x) {
+      any(is.na(x))
+    })))
+    if (type != "onesampleTTest") {
+      dataset <- jaspBase::excludeNaListwise(dataset, vars)
+    }
+    if ((type == "anova" || type == "ancova") && options[["fixedFactors"]] != "") {
+      if (any(grepl(pattern = " ", x = base::levels(dataset[, options[["fixedFactors"]]])))) {
```
[the remark this answers is not in the document]
Comment (koenderks) [reply R/bainCommon.R:63]: Bain only allows for one factor per analysis

File: R/bainCommon.R [hunk 2]
```diff
@@ -52,23 +52,17 @@
...
-      if (type == "onesampleTTest") { # For the one sample t test we do not remove the NA's listwise
-        dataset <- .readDataSetToEnd(columns.as.numeric = numerics, columns.as.factor = factors)
-      } else {
-        dataset <- .readDataSetToEnd(columns.as.numeric = numerics, columns.as.factor = factors, exclude.na.listwise = vars)
-      }
-      if ((type == "anova" || type == "ancova") && options[["fixedFactors"]] != "") {
-        if (any(grepl(pattern = " ", x = levels(dataset[, options[["fixedFactors"]]])))) {
-          jaspBase:::.quitAnalysis(gettext("Bain does not accept factor levels that contain spaces. Please remove the spaces from your factor levels to continue."))
-        }
+    dataset[factors] <- lapply(dataset[factors], as.factor)
```
[the remark this answers is not in the document]
Comment (koenderks) [reply R/bainCommon.R:55]: If this is done in JASP automatically, then it is only for the unit tests yes

File: R/bainCommon.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (vandenman) [reply R/bainCommon.R:63]: sure, but if factor levels are encoded you never need to check for this anyway :)

**Review by vandenman — APPROVED** [review]

**Pull request merged**